resources/js/script.js

```diff
@@ -26,145 +28,173 @@ $(".add-to-cart").click(function(event){
+			this.cart[i].count += count;
+			this.saveCart();
 			return;
 		}
 	}
-	var item = new Item(name, price, count);
-	cart.push(item);
-	saveCart();
+	var item = new this.Item(name, price, count);
+	this.cart.push(item);
+	this.saveCart();
 };
 
-// removeItemFromCart(name) // Removes one item
-
-function removeItemFromCart(name) {
-	for (var i in  cart) {
-		if (cart[i].name === name) {
-			cart[i].count --;
-			if (cart[i].count === 0) {
-				cart.splice(i, 1); //it removes "1" item from the array
+shoppingCart.removeItemFromCart = function(name) {
+	for (var i in  this.cart) {
+		if (this.cart[i].name === name) {
+			this.cart[i].count --;
+			if (this.cart[i].count === 0) {
+				this.cart.splice(i, 1); //it removes "1" item from the array
 			}
-			return;
+			break;
 		}
 	}
-	saveCart();
-}
-
-// removeItemFromCartAll(name) //Removes all items with name
+	this.saveCart();
+};
 
-function removeItemFromCartAll(name) {
-	for (var i in cart) {
-		if (cart[i].name === name) {
-			cart.splice(i, 1);
+shoppingCart.removeItemFromCartAll = function(name) {
+	for (var i in this.cart) {
+		if (this.cart[i].name === name) {
+			this.cart.splice(i, 1);
 			break;
 		}
 	}
-	saveCart();
-}
-
-
-//clearCart()
-
-function clearCart() {
-	cart = [];
-	saveCart();
-}
+	this.saveCart();
+};
 
-//countCart() -> Returns total count
+shoppingCart.clearCart = function() {
+	this.cart = [];
+	this.saveCart();
+};
 
-function countCart() {
+shoppingCart.countCart = function() {
 	var totalCount = 0;
-	for (var i in cart){
-		totalCount += cart[i].count;
+	for (var i in this.cart){
+		totalCount += this.cart[i].count;
 	}
 	return totalCount;
-}
-
-//totalCart() -> Returns the total cost
+};
 
-function totalCart() {
+shoppingCart.totalCart = function() {
 	var totalCost = 0;
-	for (var i in cart) {
-		totalCost += cart[i].price * cart[i].count;
+	for (var i in this.cart) {
+		totalCost += this.cart[i].price * this.cart[i].count;
 	}
-	return totalCost;
-}
-
-//listCart() -> an array of items
+	return totalCost.toFixed(2);
+};
 
-function listCart() {
+shoppingCart.listCart = function() {
 	/*return cart.slice(); -> here, the slice property makes a copy of the array; but still reference the objects inside*/
 	var cartCopy = [];
-	for (var i in cart) {
-		var item = cart[i];
+	for (var i in this.cart) {
+		var item = this.cart[i];
 		var itemCopy = {};
 		for (var p in item) {
 			itemCopy[p] = item[p];
 		}
+		itemCopy.total = (item.price * item.count).toFixed(2);
 		cartCopy.push(itemCopy);
 	}
 	return cartCopy;
-}
+};
 
-//saveCart() -> Local
+shoppingCart.saveCart = function() {
+	localStorage.setItem("shoppingCart", JSON.stringify(this.cart));
+};
 
-function saveCart() {
-	localStorage.setItem("shoppingCart", JSON.stringify(cart));
-}
+shoppingCart.loadCart = function() {
+	this.cart = JSON.parse(localStorage.getItem("shoppingCart"));
+};
+
+shoppingCart.loadCart();
+displayCart();
+
+//when you use the sintax below, it actually generates an object
```

Comment:
remove the code if it's not used - you can always look in git history to get it back. Less code = better code :-)

Reply:
👍

Reply:
I'll do that